### PR TITLE
Add Datetime Transformation

### PR DESCRIPTION
**Pull request by bmitchinson**
## Reminders:

- [x] If changing elixir code in an "app", did you update the relevant version
      in `mix.exs`?
- [ ] If altering an API endpoint, was the relevant postman collection updated?

### CODE REVIEW

File: apps/transformers/lib/transformations/date_time.ex [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    else
+      {:error, reason} ->
+        {:error, reason}
+
+      nil_payload ->
+        {:ok, nil_payload}
+    end
+  end
+
+  defp parseTime(dateString, dateFormat, sourceField) do
```
Comment (christinepoydence) [inline apps/transformers/lib/transformations/date_time.ex:26]: This function is called `parseTime`, but it accepts a `dateString` and a `dateFormat`. Should it be renamed `parseDate`?

File: apps/transformers/lib/transformations/date_time.ex [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    else
+      {:error, timexReason} ->
+        {:error,
+         "Unable to parse datetime from \"#{sourceField}\" in format \"#{dateFormat}\": #{
+           timexReason
+         }"}
+    end
+  end
+
+  defp formatDateTime(dateTime, format) do
```
Comment (christinepoydence) [inline apps/transformers/lib/transformations/date_time.ex:38]: super tiny thing - I think that elixir convention for functions is snake_case?

**Review by christinepoydence — APPROVED** [review]
This is great! General question about the datetime transformations - would it be worthwhile to document that the parsing and formatting uses the hex package Timex, so that users know where they can find documentation on acceptable values? (Maybe returned as part of the error message if there's a parsing failure?) Or are we planing to include that information on the UI?

**Comment by bmitchinson** [comment]
> This is great! General question about the datetime transformations - would it be worthwhile to document that the parsing and formatting uses the hex package Timex, so that users know where they can find documentation on acceptable values? (Maybe returned as part of the error message if there's a parsing failure?) Or are we planing to include that information on the UI?

I believe when discussed using Timex we decided that information would be in the UI.

Open to document it in other places though. What were you thinking when you mentioned documenting it? The readme of Transformations?

File: apps/transformers/lib/transformations/date_time.ex (hunk 1 above)
Comment (bmitchinson) [reply apps/transformers/lib/transformations/date_time.ex]: good call 👍  changed

File: apps/transformers/lib/transformations/date_time.ex (hunk 2 above)
Comment (bmitchinson) [reply apps/transformers/lib/transformations/date_time.ex]: glad you mentioned this, my JS is showing. Fixed!

Comment (christinepoydence) [reply apps/transformers/lib/transformations/date_time.ex]: Hah same, I find myself doing this all the time too!

**Comment by christinepoydence** [comment]
> > This is great! General question about the datetime transformations - would it be worthwhile to document that the parsing and formatting uses the hex package Timex, so that users know where they can find documentation on acceptable values? (Maybe returned as part of the error message if there's a parsing failure?) Or are we planing to include that information on the UI?
> 
> I believe when discussed using Timex we decided that information would be in the UI.
> 
> Open to document it in other places though. What were you thinking when you mentioned documenting it? The readme of Transformations?

I'm good with putting the information in the UI when we get there! Yeah, I guess the README would be the other option, but it was pretty easy for me to just check the mix.exs file to find the package, so it probably isn't necessary.

**Pull request merged**